src/main/java/uk/gov/hmcts/cmc/claimstore/events/EventProducer.java

```diff
@@ -27,22 +31,31 @@
 @Component
 public class EventProducer {
     private final ApplicationEventPublisher publisher;
+    private final boolean asyncEventOperationEnabled;
 
-    public EventProducer(ApplicationEventPublisher publisher) {
+    @Autowired
+    public EventProducer(
+        ApplicationEventPublisher publisher,
+        @Value("feature_toggles.async_event_operations_enabled") String asyncEventOperationEnabled
+    ) {
         this.publisher = publisher;
+        this.asyncEventOperationEnabled = Boolean.getBoolean(asyncEventOperationEnabled);
```

Comment:
You can use this as `boolean` type in parameter, you won't need to convert here `Boolean.getBoolean(..)`